### PR TITLE
Enable delay signing on NuGetGallery.Core

### PR DESCRIPTION
**Pull request by joelverhagen**
This is required for jobs.

Also, update to latest build tools.

### CODE REVIEW

File: build.ps1 [hunk 1]
```diff
@@ -68,28 +69,25 @@ Invoke-BuildStep 'Restoring solution packages' { `
     -ev +BuildErrors
     
 Invoke-BuildStep 'Set version metadata in AssemblyInfo.cs' { `
-    param($Path, $Version, $Branch, $Commit)
-    Set-VersionInfo -Path $Path -Version $Version -Branch $Branch -Commit $Commit `
+        $Path = Join-Path $PSScriptRoot "src\NuGetGallery\Properties\AssemblyInfo.g.cs"
```
Comment (scottbommarito) [inline build.ps1]: Not really related to this PR, but I think you can write this as

```
Invoke-BuildStep 'Set version metadata in AssemblyInfo.cs' { Set-VersionInfo } `
    -args (Join-Path $PSScriptRoot "src\NuGetGallery\Properties\AssemblyInfo.g.cs"), $SimpleVersion, $Branch, $CommitSHA `
    -ev +BuildErrors
```
and I think it looks slightly cleaner.

You could also make a function that accepts a list of paths and then calls `Set-VersionInfo` on them and apply the same trick to it.

File: build.ps1 [hunk 2]
```diff
@@ -68,28 +69,25 @@ Invoke-BuildStep 'Restoring solution packages' { `
     -ev +BuildErrors
     
 Invoke-BuildStep 'Set version metadata in AssemblyInfo.cs' { `
```
Comment (scottbommarito) [inline build.ps1]: Should probably change this build step name and the one below to indicate for which assembly the version metadata is being changed (unless you merge the two into a single step).

Comment (joelverhagen) [inline build.ps1]: sure

File: build.ps1 (hunk 1 above)
Comment (joelverhagen) [inline build.ps1]: sure

File: build.ps1 (hunk 2 above)
Comment (joelverhagen) [inline build.ps1]: Fixed by merging.

File: build.ps1 (hunk 1 above)
Comment (joelverhagen) [inline build.ps1]: Added loop.

**Pull request merged**